src/Dapr.Actors/Client/ActorProxy.cs

```diff
@@ -41,6 +43,14 @@ protected internal ActorProxy()
 
         internal IActorMessageBodyFactory ActorMessageBodyFactory { get; set; }
 
+        /// <summary>
+        /// The <see cref="IActorProxyFactory"/> to use when creating new <see cref="ActorProxy"/> instances
+        /// </summary>
+        public static void ConfigureDefaultFactory(Action<ActorProxyFactory> configureDelegate)
+        {
+            configureDelegate.Invoke(ActorProxyFactory);
+        }
+
```

Comment:
Sorry for just catching up on this, I haven't really played much with the actors code. 

We should not introducing statics that mutate configuration like this - having a static method to configure something is the same as a static property. It's not testable and it doesn't compose with libraries.

Reply:
@rynowak I agree - the only problem is that the ActorProxy uses the static method `Create` which calls the static property `ActorProxyFactory`. The only way I can think of getting the custom JSON settings to _all_ instantiated `ActorProxy`s would be to change the static `ActorProxyFactory` field a property w/ a setter on it to allow the user to set their own Actor Factory with the custom settings when the service starts up. Either way it's _still_ using a static. Short of rewriting the way that an `ActorProxy` is created I'm not sure there is a good way to get away from the statics.

If you think we shouldn't do this for _all_  ActorProxy instances I could just add an overloaded method for `Create` that takes `JsonSerializerOptions`. That would remove the need to configure the static `ActorProxyFactory`

Reply:
@LukePammant - sorry for the delay, I was travelling

Let's try to nail down an API proposal then. There's a few ideas that would work here (can be tweaked):

(sorry my samples will be C# not F# - it's out of ignorance on my part 😎)

**Option 1: create and customize the factory**

```C#
var factory = new ProxyFactory() { SerializerOptions = ... }; // customize
var proxy = factory.Create<IMyActorType>(id: ...);
```

Old API still works and uses a default factory with default options:

```C#
var = ActorProxy.Create<IMyActorType>(id: ...)
```

If you really like statics just make your own.

```C#
class MyFactory
{
    public static readonly ActorProxyFactory Instance = new ActorProxyFactory()
    {
        // customize
    }
}

// elsewhere 
var proxy = MyFactory.Instance.Create<IMyActorType>(id: ...);
```

**Option 2: additional argument**

```C#
var options = new ActorOptions() { new JsonSerializerSettings() {  ... } }; // customize
var proxy = ActorProxy.Create<IMyActorType>(id: ..., options);
```

Common usage would probably be to end up with your static own options instance.

```C#
class MyOptions
{
    public static readonly ActorOptions Instance = new ActorOptions() 
    {
         new JsonSerializerSettings() {  ... }  // customize
     };
}

// elsewhere 
var proxy = ActorProxy.Create<IMyActorType>(id: ..., MyOptions.Instance);
```

The reason for the `ActorOptions` class to future proofing. If we add more options to creating a proxy then we don't have to change the signature. 

**Option 3: Both**

Support both an argument (optional) to create an actor and the factory as a standalone class.

-----

The grow up of this is that we should also probably add `Create` functionality to ActorService. This makes it really easy to create a proxy to another actor (from inside an actor) without having to keep track of statics - you can configure it in a single place.

Reply:
My latest commit has these changes in it. I reworked a lot of the original code I submitted and think it's a little more inline with your thoughts. I didn't remove the existing statics in order to not break existing code.